File: swExpertAcademy/D2/1285.cpp
#include<iostream>

using namespace std;

int main(int argc, char **argv) {
    int test_case;
    int T;
    int n, minD, cnt, x;
    int *A;

    cin >> T;
    for (test_case = 1; test_case <= T; ++test_case) {
        minD = 100000, cnt = 0;
        cin >> n;
        A = new int(n);
        for (int i = 0; i < n; i++) {
            cin >> A[i];
            x = (A[i] > 0?A[i]:-1*A[i]);
            -1 * A[i];
            if (minD > x) {
                minD = x, cnt = 1;
            } else if (minD == x) {
                cnt += 1;
            }
        }
        cout << "#" << test_case << " " << minD <<" "<< cnt << endl;
    }
    return 0;
}

File: swExpertAcademy/D2/1285.py
T = int(input())
for test_case in range(1, T + 1):
    n = int(input())
    minD,cnt = 100000,0
    for x in list(map(int, input().split())):
        if minD > abs(x):
            minD,cnt = abs(x),1
        elif minD == abs(x):
            cnt += 1
    print(f"#{test_case} {minD} {cnt}")
